### ISSUE TITLE
Can not use hooks with html-react-parser.

### ISSUE THREAD

**Issue by mark-casias** [remarkablemark/html-react-parser#105]
Since this requires react-dom-core, which has a dependency of react@15.6.2, when trying to use any component with html-react-parser, there is a fatal error because of the duplicate React. I really am not sure how to solve this since there is no new release of react-dom-core.

**Comment by mark-casias** [comment]
PS: Opened an issue at react-dom-core asking to update as well

**Comment by remarkablemark** [comment]
So to provide some context on why this library is dependent on `react-dom-core`, React 15 used to expose all the internal lib modules but in 16 they stopped exposing them (see #43).

This library depends on several of `react-dom`'s internal modules such as `HTMLDOMPropertyConfig` and `SVGDOMPropertyConfig`, which are used when converting HTML attributes to React attributes.

So my question is were you unable to install `react@16` in your project root first?

What I mean by this is if your dependency tree for `react` looks like this:
```sh
$ npm ls react
├─┬ html-react-parser@0.7.0
│ └─┬ react-dom-core@0.0.4
│   └── react@15.6.2  extraneous
└── react@16.8.6
```

Then given node's resolution algorithm, shouldn't it pick up `react@16.8.6` first?

To reproduce the dependency tree above, install `react@latest` before `html-react-parser`:
```sh
$ rm -rf node_modules
$ npm i react html-react-parser
```

**Comment by mark-casias** [comment]
Yes, I have react@16.8.6 installed at the root. This caused the error with hooks because multiple versions of react are now referenced. This is actually the third noted reason for the error in https://reactjs.org/warnings/invalid-hook-call-warning.html

**Comment by remarkablemark** [comment]
Gotcha, thanks for letting me know. Would it be convenient for you to create a basic repo that reproduces the errors? It would be useful for testing an approach I'm investigating right now.

**Comment by remarkablemark** [comment]
@mark-casias I just released and published `html-react-parser@0.7.1`:
```sh
# npm
npm i html-react-parser@0.7.1

# yarn
yarn add html-react-parser@0.7.1
```

Can you confirm if the error is still occurring for you?

**Comment by mark-casias** [comment]
@mark-casias Thanks for the quick turn around on this. I have updated to 0.7.1 and am no longer showing multiple versions of react and hooks are not failing me.
